libtenzir/include/tenzir/detail/fanout_counter.hpp

```diff
@@ -99,4 +100,24 @@ auto make_fanout_counter(size_t expected, Continuation&& then,
                                    std::forward<ErrorContinuation>(error));
 }
 
+// template <typename State>
+// auto make_fanout_counter(size_t expected, caf::typed_response_promise<State>
+// rp) {
+//   return make_fanout_counter<State>(expected,
+//     [rp](State&& state) mutable { rp.deliver(std::move(state)); },
+//     [rp](State&&, caf::error&& e) mutable { rp.deliver(e); });
+// }
+
+inline auto
+make_fanout_counter(size_t expected, caf::typed_response_promise<void> rp) {
```

Comment:
```suggestion
make_fanout_counter(size_t expected, const caf::typed_response_promise<void>& rp) {
```

Saves one copy, since you copy twice below.